File: ChildViewControllerHelper/ChildViewControllerHelper.m
/*
   Helper function to embed/remove view controller as a child view controller. (Objective-C)

 */


#import <UIKit/UIKit.h>

void EmbedChildViewController(UIViewController *child, UIViewController *parent, UIView *containerView)
{
    [parent addChildViewController:child];
    child.view.frame = containerView.bounds;
    [containerView addSubview:child.view];
    [child didMoveToParentViewController:parent];
}

void RemoveChildViewController(UIViewController *child)
{
    [child willMoveToParentViewController:nil];
    [child.view removeFromSuperview];
    [child removeFromParentViewController];
}

File: ChildViewControllerHelper/ChildViewControllerHelper.swift
// Helper function to embed/remove view controller as a child view controller. (Swift)
//

import UIKit

func EmbedChildViewController(child: UIViewController, parent: UIViewController, containerView: UIView) {
    parent.addChildViewController(child)
    child.view.frame = containerView.bounds
    containerView.addSubview(child.view)
    child.didMoveToParentViewController(parent)
}

func RemoveChildViewController(child: UIViewController) {
    child.willMoveToParentViewController(nil)
    child.view.removeFromSuperview()
    child.removeFromParentViewController()
}
